src/coreclr/System.Private.CoreLib/src/System/Reflection/Associates.cs

```diff
@@ -50,13 +50,15 @@ internal static bool IncludeAccessor(MethodInfo? associate, bool nonPublic)
 
             bool isInherited = declaredType != reflectedType;
 
-            IntPtr[]? genericArgumentHandles = null;
+            Span<IntPtr> genericArgumentHandles = stackalloc IntPtr[0];
```

Comment:
Why not initialize to `default`? Does `stackalloc`ing zero items have any advantages?

Reply:
> Why not initialize to default?

Because then it won't compile :)

This has to do with the C# compiler's rules around ref lifetime.  When a span local is initialized to `default`, that's not tying it to any particular lifetime, which means the compiler will allow that local to escape (you could return it).  But that also means the compiler will then prevent assigning anything into it that has a more narrow lifetime, like the result of a stackalloc.  So in order to be able to later assign a stackalloc to this local, it needs to be created with local scope. `stackalloc T[0]` is equivalent to `default` but with the right lifetime.  A bit of a hack, but we use it in many places because it's the only way to achieve this short of using unsafe code.